lib/posix/semaphore.c

```diff
@@ -148,3 +213,159 @@ int sem_wait(sem_t *semaphore)
 	(void)k_sem_take(semaphore, K_FOREVER);
 	return 0;
 }
+
+sem_t *sem_open(const char *name, int oflags, ...)
+{
+	va_list va;
+	mode_t mode;
```

Comment:
I remember this was an issue on `__sparc__` so yea, definitely valid.

Instead of `mode_t` here, you might need to use `PROMOTED_MODE_T`, but it's hard to say for me. Look to see who made the commit that added it, and add them to review.

In any case, this is a really cool feature. It kind of gets you thinking about other things, like mmap, and pipe to name a couple.